test/lib/tasks/minify.js

```diff
@@ -148,3 +152,94 @@ test();`;
 		"Third taskUtil.setTag call with expected arguments");
 });
 
+test("integration: minify (without taskUtil)", async (t) => {
+	const {reader, writer, workspace} = createWorkspace();
+	const content = `
+function test(paramA) {
+	var variableA = paramA;
+	console.log(variableA);
+}
+test();`;
+	const testResource = resourceFactory.createResource({
+		path: "/test.js",
+		string: content
+	});
+	await reader.write(testResource);
+
+	await minify({
+		workspace,
+		options: {
+			pattern: "/test.js"
+		}
+	});
+
+	const expected = `function test(t){var o=t;console.log(o)}test();`;
+	const res = await writer.byPath("/test.js");
+	if (!res) {
+		t.fail("Could not find /test.js in target locator");
+	}
+	t.deepEqual(await res.getString(), expected, "Correct file content");
+
+	const resDbg = await writer.byPath("/test-dbg.js");
+	if (!resDbg) {
+		t.fail("Could not find /test-dbg.js in target locator");
+	}
+	t.deepEqual(await resDbg.getString(), content, "Correct debug-file content");
+
+	const expectedSourceMap =
+		`{"version":3,"sources":["test-dbg.js"],"names":["test","paramA","variableA","console","log"],` +
+		`"mappings":"AACA,SAASA,KAAKC,GACb,IAAIC,EAAYD,EAChBE,QAAQC,IAAIF,GAEbF","file":"test.js"}`;
+
+	const resSourceMap = await writer.byPath("/test.js.map");
+	if (!resSourceMap) {
+		t.fail("Could not find /test-dbg.js.map in target locator");
+	}
+	t.deepEqual(await resSourceMap.getString(), expectedSourceMap, "Correct source map content");
+});
+
+test("integration: minify omitSourceMapResources=false (without taskUtil)", async (t) => {
+	const {reader, writer, workspace} = createWorkspace();
+	const content = `
+function test(paramA) {
```

Comment:
should we add there some indentation?

Reply:
answered above https://github.com/SAP/ui5-builder/pull/709#discussion_r823602439